### DISCUSSION TITLE
Pancreas example dataset

### DISCUSSION THREAD

**Discussion by parashardhapola** [theislab/scvelo#474] [Q&A] [answered]
Hi,

I need some help interpreting the analysis of the pancreas dataset. I followed the tutorial/recipe here: https://scvelo.readthedocs.io/Pancreas.html and got the same velocity embedding stream graph as the documentation (also shown below)

![image](https://user-images.githubusercontent.com/10581156/93807813-75f5a100-fc4b-11ea-8e70-1d8ed3b9350a.png)

I modified the recipe by removing some precomputed data, as shown in the code snippet below:

```
import h5py
import scvelo as scv
import scanpy as sc

adata = scv.datasets.pancreas()

h5 = h5py.File('./data/Pancreas/endocrinogenesis_day15.h5ad', mode='r+')
del h5['uns']['neighbors']
del h5['uns']['pca']
del h5['obsm']['X_pca']
del h5['obsm']['X_umap']

adata = scv.read('./data/Pancreas/endocrinogenesis_day15.h5ad')

scv.pp.filter_and_normalize(adata, min_shared_counts=1, n_top_genes=2000)
scv.pp.moments(adata, n_pcs=30, n_neighbors=30)
sc.tl.umap(adata)
scv.tl.velocity(adata)
scv.tl.velocity_graph(adata)
scv.pl.velocity_embedding_stream(adata, basis='umap')
```

![image](https://user-images.githubusercontent.com/10581156/93810610-3630b880-fc4f-11ea-80ca-a20911aef9d4.png)

The fate of the **epsilon** has changed quite substantially and the velocity embedding stream now leads into **alpha** cells.

From the paper, I understand that the stochastic steady-state model might sometimes perform inaccurately and it might be a good idea to run the dynamical model to get an overall improved accuracy of velocities across the manifold. So I reran the code using the dynamical model.

```
import h5py
import scvelo as scv
import scanpy as sc

# Deleted existing h5ad file beforehand while in jupyter notebook using the following command:
# !rm -rf ./data/Pancreas

adata = scv.datasets.pancreas()

h5 = h5py.File('./data/Pancreas/endocrinogenesis_day15.h5ad', mode='r+')
del h5['uns']['neighbors']
del h5['uns']['pca']
del h5['obsm']['X_pca']
del h5['obsm']['X_umap']

adata = scv.read('./data/Pancreas/endocrinogenesis_day15.h5ad')

scv.pp.filter_and_normalize(adata, min_shared_counts=20, n_top_genes=2000)
scv.pp.moments(adata, n_pcs=30, n_neighbors=30)
sc.tl.umap(adata)

scv.tl.recover_dynamics(adata)
scv.tl.velocity(adata, mode='dynamical') 
scv.tl.velocity_graph(adata)
scv.pl.velocity_embedding_stream(adata, basis='umap')
```
The velocities now show a "backflow" in the **alpha** cells which cannot be seen in either the documentation or the paper. 

![image](https://user-images.githubusercontent.com/10581156/93812832-70e82000-fc52-11ea-8b68-5c632d8eefd6.png)

Is this due to a difference in UMAP embedding that we see this or maybe due to changes in parameter values for neighbor calculation? Or have I messed up somewhere?

I would really appreciate your comments on this.

Versions from `scv.logging.print_versions()`
```
scvelo==0.2.2  scanpy==1.5.1  anndata==0.7.1  loompy==2.0.17  numpy==1.19.1  scipy==1.5.2  matplotlib==3.3.1  sklearn==0.23.2  pandas==1.1.1
```

Thanks!

**Comment by VolkerBergen** [comment, marked as answer]
The projection of velocities depends on the embedding coordinates and the genes that are included in the analysis. In the second example you're filtering out many important genes (setting `min_shared_counts=1` and then going for 2k hvg's is not a good selection criteria - the min_counts definitely needs to be higher, otherwise many important genes are removed while secondary important ones with few counts and high variations are included for velocity analysis even though they simply cannot yield any signal in splicing kinetics with such low cell numbers). Anyways, I wouldn't claim from a 2D projection, that the fate of the epsilon cells have changed. In fact, if you explore velocities in their high dimension and for instance run Markov chain forward starting from epsilon cells, you'd also find them to be moving towards the alpha fate, in both examples. 

For the latter dynamical example, this is due to gene selection. It is more likely to filter out important genes when setting the thresholds relatively low (min counts) and cutting out all those that don't make it into the top2000 hvg. I would generally recommend to set the threshold `min_shared_counts` relatively high (at least 30) and hvg rather low. In anyway, as you see, this is highly sensitive to gene selection, thus for interpretation it is always critical to look into gene-wise dynamics.
